packages/ubuntu_desktop_installer/lib/utils/version_extractor.dart

```diff
@@ -0,0 +1,65 @@
+import 'dart:io';
+
+class VersionExtractor {
+  static VersionExtractor? _instance;
+
+  static VersionExtractor get instance {
+    if (_instance == null) {
+      _instance = VersionExtractor._();
+    }
+    return _instance!;
+  }
+
+  VersionExtractor._();
+
+  String loadedVersion = '';
+
+  String _dummy() {
+    return 'Ubuntu';
+  }
+
+  String extractLocalVersion(String content) {
+    return content.substring(content.indexOf("\"") + 1, content.length - 1);
+  }
+
+  String getVersion({
+    String path = '/cdrom/.disk/info',
+    String localPath = '/etc/os-release',
+  }) {
+    if (loadedVersion.isNotEmpty) {
+      return loadedVersion;
+    }
+
+    final fileOnCdrom = File(path);
+    final localVersionFile = File(localPath);
+    if (fileOnCdrom.existsSync()) {
+      try {
+        final content = fileOnCdrom
+            .readAsLinesSync()
+            .firstWhere((line) => line.trim().isNotEmpty);
+
+        loadedVersion = extractVersion(content);
+      } catch (e) {
+        //TODO: configure logger?
```

Comment:
This doesn't do what you want: when catching an exception here (and lower down in the same method), the code should move on to trying the next extraction method for the product info, and as a last resort use the fallback hardcoded string. Here it will simply return an empty `loadedVersion`.

I don't think there's a need for logging here.

The tests for the class should try and test all these cases.